dts/bindings/pinctrl/st,stm32f1-pinctrl.yaml

```diff
@@ -11,6 +11,31 @@ properties:
     reg:
       required: true
 
+    swj-cfg:
+        type: string
+        required: false
+        description: |
+          Configures number of pins assigned to the SWJ debug port.
+
+          +-----------------+-----------------------------------------------+
+          | Value           | Description                                   |
+          +-----------------+-----------------------------------------------+
+          | no-njtrst       | Full SWJ (JTAG-DP + SW-DP) but without NJTRST.|
+          |                 | Releases PB4.                                 |
+          +-----------------+-----------------------------------------------+
+          | jtag-disable    | JTAG-DP Disabled and SW-DP Enabled.           |
+          |                 | Releases PA15 PB3 PB4.                        |
+          +-----------------+-----------------------------------------------+
+          | disable         | JTAG-DP Disabled and SW-DP Disabled           |
+          |                 | Releases: PA13 PA14 PA15 PB3 PB4.             |
+          +-----------------+-----------------------------------------------+
+
+          If absent, then Full SWJ (JTAG-DP + SW-DP) is used (reset state).
+        enum:
+          - "no-njtrst"
```

Comment:
I wonder if we should add "enable" as a possible value. This is not strictly required, but this may add in clarity.
Not blocking, I'm just opening the discussion.

Reply:
Added `full` as the first value and made it `default`. 
Used `full` instead of `enable` as it refers to the `swj_cfg` " Full SWJ (JTAG-DP + SW-DP)".